release/models/mirror/openconfig-mirror.yang

```diff
@@ -0,0 +1,241 @@
+module openconfig-mirror {
+
+  yang-version "1";
+
+  // namespace
+  namespace "http://openconfig.net/yang/mirror";
+
+  prefix "oc-mirror";
+
+  // import some basic types
+  import openconfig-extensions {
+    prefix oc-ext;
+  }
+  import openconfig-interfaces {
+    prefix oc-if;
+  }
+  import openconfig-acl {
+    prefix oc-acl;
+  }
+  import openconfig-inet-types {
+    prefix oc-inet;
+  }
+  import openconfig-network-instance {
+    prefix oc-ni;
+  }
+
+  // meta
+  organization
+    "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+     www.openconfig.net";
+
+  description
+    "Model for configuration and operational state data related
+     to capturing packets and sending them to some destination
+     for debugging or monitoring.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2024-06-12" {
+    description
+      "Initial revision";
+    reference
+      "0.1.0";
+  }
+
+  // extension statements
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  // feature statements
+  // identity statements
+  // typedef statements
+  typedef mirror-action {
+    type enumeration {
+      enum FILE {
+        description
+          "Captured packets are stored in a local file on the
+           device in pcap format.";
+      }
+      enum GRE {
```

Comment:
I'm wondering if this is the best approach. 

There could be other encapsulations (e.g. GUE, VXLAN) that can share a set of properties (source-ip, network-instance, etc).

Additionally, the blanket "GRE" enum does not cover different parameters that can be associated with a GRE tunnel and/or GRE flavors. For example, it can be l2 over gre or l3 over gre. GRE tunnel itself can have additional parameters such as the presence of the `key` field in the header.

Perhaps it would be more convenient to use enum `tunnel` or `remote` and then create another grouping that can be used to define the tunnel parameters.